src/main/java/com/rusticisoftware/tincan/RemoteLRS.java

```diff
@@ -223,19 +220,113 @@ public void onHeaders(Response webResp) {
+
+            response.setStatus(httpResponse.getStatus());
+            response.setStatusMsg(httpResponse.getReason());
+            for (HttpField header : httpResponse.getHeaders()) {
+                response.setHeader(header.getName(), header.getValue());
             }
 
-            if (req.getContent() != null) {
-                webReq.content(new BytesContentProvider(req.getContent()));
+
+
+            // Use try-with-resources to close input stream.
+            try (InputStream responseContent = listener.getInputStream())
+            {
+
+                if (response.getContentType() != null && response.getContentType().contains("multipart/mixed")) {
+
+                    MultipartParser responseHandler = new MultipartParser(responseContent);
+
+                    // We need to get the first part that contains the statements and parse them
+                    responseHandler.nextPart();
+
+                    ArrayList<Statement> statements = new ArrayList<Statement>();
+
+                    if (responseHandler.getHeaders().get("Content-Type").contains("application/json")) {
+                        JsonNode statementsNode = (new StringOfJSON(new String(responseHandler.getContent())).toJSONNode());
+                        if (! (statementsNode.findPath("statements").isMissingNode())) {
+                            statementsNode = statementsNode.findPath("statements");
+                            for (JsonNode obj : statementsNode) {
+                                statements.add(new Statement(obj));
+                            }
+                        }
+                        else {
+                            statements.add(new Statement(statementsNode));
+                        }
+                    }
+
+                    while (! responseHandler.noMoreParts) {
+                        responseHandler.nextPart();
+                        String hashToMatch = responseHandler.getHeaders().get("X-Experience-API-Hash");
+                        for (Statement stmt : statements) {
+                            if (stmt.getAttachments() != null) {
+                                for (Attachment a : stmt.getAttachments()) {
+                                    if (a.getSha2().equals(hashToMatch)) {
+                                        a.setContent(responseHandler.getContent());
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    StatementsResult responseStatements = new StatementsResult();
+                    responseStatements.setStatements(statements);
+                    response.setContentBytes(responseStatements.toJSONNode(TCAPIVersion.V101).toString().getBytes());
+                }
+                else {
+                    String responseContentString = IO.toString(responseContent);
```

Comment:
I don't think we were stringifying response bodies before and this seemingly wouldn't work for binary documents. Is there a reason for this change?